src/nupic/algorithms/Connections.hpp

```diff
@@ -402,6 +402,15 @@ class Connections : public Serializable
    * @param inputVector  An SDR
    * @param increment  Change in permanence for synapses with active presynapses.
    * @param decrement  Change in permanence for synapses with inactive presynapses.
+   *
+   * @params previousUpdates & currentUpdates contain the permanence changes done
+   * by this method.  AdaptSegment will not  apply the same learning update to a
+   * synapse on consequetive cycles, because then staring at the same object for
+   * too long will mess up the synapses. This change allows it to work with
+   * timeseries data which moves very slowly, instead of the usual HTM inputs
+   * which reliably change every cycle. See also (Kropff & Treves, 2008).
+   *    - These vectors will be resized to "con.synapseFlatListLength()".
+   *    - To reset between time-series sequences: zero or clear the vector.
```

Comment:
> To reset between time-series sequences: zero or clear the vector. 

this is OK, as it's documented, but not an ideal API. 
Can the 2 vectors be removed from method's arguments, and instead be Connection's private members? (+argument bool ignorePrevPatterns) ?

Reply:
@breznak,
I did some thinking about the API and here is what I want to do:

1) Constructor accepts optional flag `timeseries` which enables this feature.  If this features is *not* enabled then there will be *no* performance impact.  Memory will only be allocated if this feature is used.
2) New method `reset()` for users to call in between sequences.
3) Method `computeActivity()` will move currentUpdates to previousUpdates, and clear the currentUpdates.
4) Method `adaptSegment()` will have the same API as master branch currently does.  
5) Connections class will have constant public reference to the currentUpdates and previousUpdates.

Reply:
Yes, I like this much better! :+1: 

> Connections class will have constant public reference to the currentUpdates and previousUpdates. 

consider having these as get methods, so it can throw if timeseries not used? Also computeActivity would not fill the updates at all.